utils-aio/Cargo.toml

```diff
@@ -6,12 +6,28 @@ edition = "2021"
 [lib]
 name = "utils_aio"
 
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+[features]
+default = ["muxer"]
+codec = []
+muxer = ["dep:tokio"]
 
 [dependencies]
 bytes = "1.1.0"
 prost = "0.9"
-tokio-util = { version = "0.7.1", features = ["codec"] }
+tokio = { version = "1.21.0", optional = true }
```

Comment:
Is there a good reason for specifying dependencies that tightly. My understanding was that as a general rule of thumb:
- If there is a X.x.x version, where X != 0, then  use "X" as a dependency
- If X == 0, then use "0.x" as a dependency

Otherwise we opt out of potentially useful non-breaking updates and it could make the build process more complicated when having to deal with several very similar versions of the same crate.